auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/internal/EditAccessGroupsController.kt

```diff
@@ -0,0 +1,173 @@
+package uk.gov.communities.delta.auth.controllers.internal
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.auth.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import kotlinx.serialization.Serializable
+import org.slf4j.LoggerFactory
+import uk.gov.communities.delta.auth.config.LDAPConfig
+import uk.gov.communities.delta.auth.plugins.ApiError
+import uk.gov.communities.delta.auth.repositories.isInternal
+import uk.gov.communities.delta.auth.services.*
+
+class EditAccessGroupsController(
+    private val userLookupService: UserLookupService,
+    private val groupService: GroupService,
+    private val organisationService: OrganisationService,
+    private val accessGroupsService: AccessGroupsService,
+    private val memberOfToDeltaRolesMapperFactory: MemberOfToDeltaRolesMapperFactory,
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    fun route(route: Route) {
+        route.post { updateUserAccessGroups(call) }
+    }
+
+    // This endpoint takes a list of access groups to add and to remove in the form of maps of the group name (with
+    // delta prefix) and a list of organisations for that access group. Generation of the correct lists is performed in
+    // Delta.
+    private suspend fun updateUserAccessGroups(call: ApplicationCall) {
```

Comment:
We should check a couple more things:

* The list of requested organisations is a subset of the user's domain organisations
  * Can get with something like `organisationService.findAllByDomain(emailToDomain(user.email))`
* That the organisations in the `accessGroupsRequest` are a subset of the requested organisations